node-data/src/events.rs

```diff
@@ -0,0 +1,45 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+mod blocks;
+mod transactions;
+
+pub use blocks::BlockEvent;
+pub use transactions::TransactionEvent;
+
+#[derive(Clone, Debug)]
+pub struct Event {
+    pub component: &'static str,
+    pub topic: &'static str,
+    pub entity: String,
+    pub data: EventData,
+}
+
+pub trait EventSource {
+    fn component(&self) -> &'static str;
```

Comment:
Did you consider making this `const COMPONENT: &str`? Or do you expect this would change depending on the data in the event itself?

Reply:
I don't expect it to change. So I'm going to apply your suggestion, thanks